main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1018,12 +1038,9 @@ object Typer {
         } yield (elementConstrs.flatten, Type.mkTuple(elementTypes, loc), resultEff)
 
       case KindedAst.Expression.RecordEmpty(tvar, loc) =>
```

Comment:
Do we actually need a tvar here?

Reply:
What's the rule on when we need tvars?

Reply:
Not sure. The rule is when we need to record the type for later.